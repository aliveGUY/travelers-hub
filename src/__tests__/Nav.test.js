import renderer from 'react-test-renderer';
import {
  render, fireEvent, screen,
} from '@testing-library/react';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import Navbar from '../components/Nav';

describe('Nav bar testing', () => {
  test('Nav bar Snapshot', () => {
    const tree = renderer.create(<Router><Navbar /></Router>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Nav bar rockets link', () => {
    render(<Router><Navbar /></Router>);
    const link = screen.getByText('Rockets');

    fireEvent.click(link);

    expect(window.location.pathname).toBe('/rockets');
  });

  test('Nav bar Missions link', () => {
    render(<Router><Navbar /></Router>);
    const link = screen.getByText('Missions');
    fireEvent.click(screen.getByText('My Profile'));

    fireEvent.click(link);

    expect(window.location.pathname).toBe('/');
  });

  test('Nav bar My profile link', () => {
    render(<Router><Navbar /></Router>);
    const link = screen.getByText('My Profile');

    fireEvent.click(link);

    expect(window.location.pathname).toBe('/myprofile');
  });
});
