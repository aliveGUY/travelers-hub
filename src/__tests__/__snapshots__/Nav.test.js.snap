// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav bar testing Nav bar Snapshot 1`] = `
<nav
  className="Navbar mt-3"
>
  <div
    className="navbar-Logo"
  >
    <img
      alt="Logo"
      className="navbar-img"
      src="Logo.png"
    />
    <h1
      className="mx-3"
    >
      Space Travelers' Hub
    </h1>
  </div>
  <ul
    className="navbar"
  >
    <li
      className="navbar-li"
    >
      <a
        className="nav-link"
        href="/rockets"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li
      className="navbar-li"
    >
      <a
        aria-current="page"
        className="nav-link active"
        href="/"
        onClick={[Function]}
      >
        Missions
      </a>
    </li>
    <li
      className="navbar-li"
    >
      <a
        className="nav-link"
        href="/myprofile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </li>
  </ul>
</nav>
`;
